### PR TITLE
fixes #268 Addressable::URI.heuristic_parse: uri starts with a digit …

### LINKED ISSUE #268: Error in Addressable::URI.heuristic_parse for numeric domain + port
Gem version **2.5.2**
```ruby
p Addressable::URI.heuristic_parse('127.0.0.1:8089', scheme: 'https').to_s
p Addressable::URI.heuristic_parse('random.dyndns.org', scheme: 'https').to_s
p Addressable::URI.heuristic_parse('random.dyndns.org:8089', scheme: 'https').to_s
p Addressable::URI.heuristic_parse('7777.example.org', scheme: 'https').to_s
p Addressable::URI.heuristic_parse('7777.example.org:8089', scheme: 'https').to_s
```

Output:
```
"https://127.0.0.1:8089"
"https://random.dyndns.org"
"https://random.dyndns.org:8089"
"https://7777.example.org"
/home/zurginq/.rvm/gems/ruby-2.3.1@monitoring/gems/addressable-2.5.2/lib/addressable/uri.rb:874:in `scheme=': Invalid scheme format: 7777.example.org (Addressable::URI::InvalidURIError)
```
For gem version **2.3.6** this works without errors

### PR DESCRIPTION
**Pull request by Mifrill**
…and has specified port

### CODE REVIEW

File: spec/addressable/uri_spec.rb [hunk 1]
```diff
@@ -5498,7 +5498,7 @@ def to_s
     expect(uri.tld).to eq("uk")
   end
 
-  context "witch " do
+  context "which " do
```
Comment (Mifrill) [inline spec/addressable/uri_spec.rb]: preamble: https://github.com/sporkmonger/addressable/pull/294#pullrequestreview-111029238

**Comment by Mifrill** [comment]
closed #275

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -18,4 +18,5 @@
 
 RSpec.configure do |config|
   config.warnings = true
+  config.filter_run_when_matching :focus
```
Comment (dentarg) [inline spec/spec_helper.rb]: FYI this require rspec core >= 3.5

https://relishapp.com/rspec/rspec-core/v/3-5/docs/filtering/filter-run-when-matching (switch version to 3.4 and docs can't be found)

the Gemfile here says rspec 3.0: https://github.com/sporkmonger/addressable/blob/b919fb0ad73a0eb9add6c0d3f07a6f14a414ecf5/Gemfile#L6

I think that should be bumped if this goes in

Comment (Mifrill) [reply spec/spec_helper.rb]: @dentarg  you know better what is better for this repo
Tell me to change the version of the gem or delete this line?

I will do as you say, thank you

File: lib/addressable/uri.rb [hunk 3]
```diff
@@ -194,6 +194,8 @@ def self.heuristic_parse(uri, hints={})
         uri.sub!(/^file:\/+/i, "file:///")
       when /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/
         uri.sub!(/^/, hints[:scheme] + "://")
+      else
+        uri = hints[:scheme] + "://" + uri if uri[/\A\d+\..*:\d+\z/]
```
Comment (sporkmonger) [inline lib/addressable/uri.rb]: Why not just make this another when clause?

Comment (Mifrill) [reply lib/addressable/uri.rb]: sure, it's much better, thank you

**Pull request merged**